src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -671,16 +682,22 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 #endif
 }
 	case SN_Add:
+	case SN_Divide:
 	case SN_Max:
 	case SN_Min:
 	case SN_Multiply:
 	case SN_Subtract: {
+		if (!is_element_type_primitive (fsig->ret) || !is_element_type_primitive (fsig->params [0]) || !is_element_type_primitive (fsig->params [1]))
+			return NULL;
```

Comment:
Aren't these the same types ?

Reply:
I had an impression that there was no type checks before this point. So I wanted to check all vector types to be safe.